plugin/plugin_init.go

```diff
@@ -0,0 +1,75 @@
+package plugin
+
+import (
+	"flag"
+	"syscall"
+	"os/signal"
+	"net/url"
+	"golang.org/x/oauth2"
+	"context"
+
+	"github.com/sirupsen/logrus"
+	"github.com/google/go-github/github"
+	"k8s.io/test-infra/prow/plugins"
+	"github.com/arquillian/ike-prow-plugins/plugin/server"
+	. "github.com/arquillian/ike-prow-plugins/plugin/utils"
+	"k8s.io/test-infra/prow/pluginhelp/externalplugins"
+	"strconv"
+
+	"net/http"
+)
+
+var (
+	port              = flag.Int("port", 8888, "Port to listen on.")
+	dryRun            = flag.Bool("dry-run", true, "Dry run for testing. Uses API tokens but does not mutate.")
+	pluginConfig      = flag.String("ike-plugins-config", "/etc/plugins/plugins", "Path to ike-plugins config file.")
+	githubEndpoint    = flag.String("github-endpoint", "https://api.github.com", "GitHub's API endpoint.")
+	githubTokenFile   = flag.String("github-token-file", "/etc/github/oauth", "Path to the file containing the GitHub OAuth secret.")
+	webhookSecretFile = flag.String("hmac-secret-file", "/etc/webhook/hmac", "Path to the file containing the GitHub HMAC secret.")
+)
+
+// InitPlugin instantiates logger, loads the secrets from the flags, sets context to background and starts server with
+// the attached event handler.
+func InitPlugin(
+	log *logrus.Entry,
+	createHandler func(Client *github.Client) server.GitHubEventHandler,
+	createServer func(HmacSecret []byte, testKeeperEvents server.GitHubEventHandler) *server.Server,
```

Comment:
We can make these functions types like `externalplugins.ExternalPluginHelpProvider` is. WDYT?

Reply:
good point :1st_place_medal: